modules/imgproc/src/opencl/box_filter.cl

```diff
@@ -0,0 +1,78 @@
+// This file is part of OpenCV project.
+// It is subject to the license terms in the LICENSE file found in the top-level directory
+// of this distribution and at http://opencv.org/license.html.
+
+__kernel void box_filter(__global const uint* src, int ssy, int srcOffsetX, int srcOffsetY,
+                         __global uint* dst, int dsy, int dst_offset, int rows, int cols
+#ifdef NORMALIZE
+                         , float alpha
+#endif
+                         )
+{
+    int x = get_global_id(0);
+    int y = get_global_id(1) * 2;
+    int ssx, dsx;
+
+    if ((x * 16) >= cols || y >= rows) return;
+
+    uint4 line[4];
```

Comment:
it looks vload4(vector uint4) is faster than vload16(vector uchar16). The test is conducted on Intel Broadwell GT2.